index.php

```diff
@@ -28,6 +28,52 @@
 
 $id = required_param('id', PARAM_INT);
 
-require_login();
+$course = $DB->get_record('course', ['id' => $id], '*', MUST_EXIST);
+require_course_login($course);
 
-redirect($CFG->wwwroot . '/mod/kanban/view.php?id=' . $id);
+$coursecontext = context_course::instance($course->id);
+
+$PAGE->set_url('/mod/kanban/index.php', ['id' => $id]);
+$PAGE->set_title(format_string($course->fullname));
+$PAGE->set_heading(format_string($course->fullname));
+$PAGE->set_context($coursecontext);
+
+echo $OUTPUT->header();
+
+$modulenameplural = get_string('modulenameplural', 'mod_kanban');
+echo $OUTPUT->heading($modulenameplural);
+
+$kanbans = get_all_instances_in_course('kanban', $course);
+
+if (empty($kanbans)) {
+     notice(get_string('no$kanbaninstances', 'mod_kanban'), new moodle_url('/course/view.php', ['id' => $course->id]));
+}
+
+$usesections = course_format_uses_sections($course->format);
+$table = new html_table();
+$table->attributes['class'] = 'generaltable mod_index';
+
+if ($usesections) {
+    $table->head = [get_string('sectionname', 'format_' . $course->format), get_string('name')];
+    $table->align = ['left', 'left'];
+} else {
+    $table->head  = [get_string('name')];
+    $table->align = ['left'];
+}
+
+foreach ($kanbans as $kanban) {
+    $linkcss = null;
+    if (!$kanban->visible) {
```

Comment:
Not only do you have to check the visibility, but also that the current user has the capability to view the kanban.

Hidden by restrictions for example has to be reflected here, or just in general the capability to view the kanban course module